js/controller.js

```diff
@@ -141,6 +141,7 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
       "isPlaybackReadySubscribed": false,
       "isSkipAdClicked": false,
       "isInitialPlay": false,
+      "initialPlayHasOccurred": false,
```

Comment:
Once this was set once in initialPlay, this is no longer reset on setEmbedCode or assetChanged.

Reply:
This is on purpose, the variable indicates whether `initialPlay` has ever occurred. It's basically a way of knowing whether we're dealing with the first video on embed code changed.